elide-contrib/elide-dynamic-config-helpers/src/main/java/com/yahoo/elide/contrib/dynamicconfighelpers/validator/DynamicConfigValidator.java

```diff
@@ -0,0 +1,172 @@
+        if (isSecurityConfig) {
+            String securityConfigContent = DynamicConfigHelpers
+                    .readConfigFile(new File(absoluteBasePath + DynamicConfigHelpers.SECURITY_CONFIG_PATH));
+            validateConfigForMissingVariables(securityConfigContent, variables);
+            try {
+                elideSecurityConfig = DynamicConfigHelpers.getElideSecurityPojo(absoluteBasePath, variables);
+            } catch (Exception e) {
+                throw new IllegalStateException("Error while parsing security config at location: " + absoluteBasePath,
+                        e);
+            }
+        }
+        return isSecurityConfig;
+    }
+
+    private static boolean readTableConfig(String absoluteBasePath) {
+        boolean isTableConfig = exists(absoluteBasePath + DynamicConfigHelpers.TABLE_CONFIG_PATH);
+        if (isTableConfig) {
+            Collection<File> tableConfigs = FileUtils.listFiles(
+                    new File(absoluteBasePath + DynamicConfigHelpers.TABLE_CONFIG_PATH), new String[] { "hjson" },
+                    false);
+            if (tableConfigs.isEmpty()) {
+                usage();
+                throw new IllegalStateException("No Table Configs found at location: " + absoluteBasePath
+                        + DynamicConfigHelpers.TABLE_CONFIG_PATH);
+            }
+            for (File tableConfig : tableConfigs) {
+                String tableConfigContent = DynamicConfigHelpers.readConfigFile(tableConfig);
+                validateConfigForMissingVariables(tableConfigContent, variables);
+            }
+            try {
+                elideTableConfig = DynamicConfigHelpers.getElideTablePojo(absoluteBasePath, variables);
+            } catch (Exception e) {
+                throw new IllegalStateException("Error while parsing table config at location: " + absoluteBasePath, e);
+            }
+        } else {
+            usage();
+            throw new IllegalStateException("Table Configs Directory doesn't exists at location: " + absoluteBasePath
+                    + DynamicConfigHelpers.TABLE_CONFIG_PATH);
+        }
+        return isTableConfig;
+    }
+
+    private static boolean exists(String filePath) {
+        return new File(filePath).exists();
+    }
+
+    private static void validateConfigForMissingVariables(String config, Map<String, Object> variables) {
+        Pattern regex = Pattern.compile("<%(.*?)%>");
+        Matcher regexMatcher = regex.matcher(config);
+        while (regexMatcher.find()) {
+            String str = regexMatcher.group(1).trim();
+            if (!variables.containsKey(str)) {
+                throw new IllegalStateException(str + " is used as a variable in either table or security config files "
+                        + "but is not defined in variables config file.");
+            }
+        }
+    }
+
+    private static boolean validateSqlInTableConfig(ElideTableConfig elideTableConfig) {
```

Comment:
We also need to validate the metric and column definitions.  They should not contain any semi colons - that would allow them to insert SQL statements.

Reply:
updated